add various api tests for Post endpoints

jjanickamatraszek opened the pull request.

jjanickamatraszek committed:
add various api tests for Post endpoints

jjanickamatraszek committed:
update getSelectedEntriesFromGetAllPostTest with retry

jjanickamatraszek committed:
change core_log_level to DEBUG

ankochergin commented on src/main/java/model/PostRs.java:
```diff
@@ -0,0 +1,19 @@
+@AllArgsConstructor
+@NoArgsConstructor
+@EqualsAndHashCode
+public class PostRs {
```
I think it can be just `Post`, `SearchedPost`, `DeletedPost` etc

ankochergin commented on src/main/resources/agent.properties:
```diff
@@ -0,0 +1,6 @@
+reporting.enabled=true
+reporting.projectKey=JOANNA
+reporting.server.hostname=https://solvdinternal.zebrunner.com
+reporting.server.access-token=${zebrunner_token}
```
GitHub secrets 👍

ankochergin commented on src/test/java/ApiTests.java:
```diff
@@ -0,0 +1,217 @@
+    }
+
+    @DataProvider(name = "get incorrect user ids")
+    public Object[][] getIncorrectUserIds() {
```
I suggest to make same names to avoid confusions later
```suggestion
    @DataProvider(name = "incorrectUserIds")
    public Object[][] incorrectUserIds() {
```

ankochergin commented on src/test/java/ApiTests.java:
```diff
@@ -0,0 +1,217 @@
+
+    @Test(description = "Test with response validation using JSON template")
+    @TestCaseKey(value = "JOANNA-60")
+    public void deletePost_v1Test() {
```
don't use `_` in methods names, please always follow java conventions

ankochergin commented on src/test/java/ApiTests.java:
```diff
@@ -0,0 +1,217 @@
+        SearchPostsMethod searchPostsMethod = new SearchPostsMethod(searchPhrase);
+        SearchPostsRs searchPostsRs = searchPostsMethod.callAPIExpectSuccess().as(SearchPostsRs.class);
+        searchPostsMethod.validateResponseAgainstSchema("api/posts/getAllPosts/rs.schema");
+        int amountOfReturnedPosts = searchPostsRs.getPosts().size();
```
```suggestion
        int returnedPostsCount = searchPostsRs.getPosts().size();
```

ankochergin commented on src/test/java/ApiTests.java:
```diff
@@ -0,0 +1,217 @@
+        int amountOfReturnedPosts = searchPostsRs.getPosts().size();
+        Assert.assertTrue(searchPostsRs.getPosts().size() > 0,
+                "No posts were returned for phrase '%s'".formatted(searchPhrase));
+        long amountOfPostsContainingSearchPhrase = searchPostsRs
```
```suggestion
        long gfilteredPostsCount = searchPostsRs
```

ankochergin commented on src/test/java/ApiTests.java:
```diff
@@ -0,0 +1,217 @@
+        String actualErrorMessage = response.jsonPath().getString("message");
+        Assert.assertEquals(actualErrorMessage, expectedErrorMessage,
+                "Error message is different than expected");
+    }
```
usually for each data set from data provider we have separate test case id

jjanickamatraszek committed:
PR update

- correct model classes' names
- unify data provider's name value and method name
- add separate tc ids for each data set in data provider
- refactor test methods and params' names
